src/astro/databases/google/bigquery.py

```diff
@@ -140,6 +141,23 @@ def get_merge_initialization_query(parameters: Tuple) -> str:
         """
         return "RETURN"
 
+    def _get_credentials(self) -> Optional[Credentials]:
+        conn_extra = self.hook.get_connection(self.conn_id).extra_dejson
+
+        if "extra__google_cloud_platform__keyfile_dict" in conn_extra:
+            return Credentials.from_service_account_info(
+                conn_extra["extra__google_cloud_platform__keyfile_dict"]
+            )
+
+        if "extra__google_cloud_platform__key_path" in conn_extra:
+            return Credentials.from_service_account_file(
+                conn_extra["extra__google_cloud_platform__key_path"]
+            )
+
+        # Falling back to Default Authentication Methods,
+        # see https://pandas-gbq.readthedocs.io/en/latest/howto/authentication.html for details.
```

Comment:
Can we not replace this with the following?:

```python
self.hook._get_credentials(),
```

Reply:
Yes, we can! Good catch! 🥇 

> I know it is an internal method but wdyt?

🤔 🤞 that it won't change. As it is a _private_ function it can be changed even in patch versions without a changelog note. Are you a code owner for this file in the airflow project? We somehow need to subscribe to changes so we can react quickly 😅

Reply:
Actually.. It already has a lot of references in other hooks:
<img width="637" alt="Screenshot 2022-08-18 at 07 22 31" src="https://user-images.githubusercontent.com/23282078/185300223-d9b2de65-028b-4d20-bc73-e1c2ceedac6b.png">

I think it should be public actually. WDYT? I can open a PR in airflow core.

Reply:
I have changed my code to use the existing `_get_credentials` function and have rewritten the tests.

Reply:
Btw, I have created a PR [here](https://github.com/apache/airflow/pull/25785).

Reply:
Awesome